enhancements/cluster-logging/forward_to_cloudwatch.md

```diff
@@ -0,0 +1,249 @@
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+[Amazon Cloudwatch][aws-cw] is a hosted monitoring and log storage service.
+This proposal extends the `ClusterLogForwarder` API with an output type for Cloudwatch.
+
+## Motivation
+
+Amazon Cloudwatch is a popular log store.
+We have requests from external and  Red Hat-internal customers to support it.
+
+### Goals
+
+Enable log forwarding to Cloudwatch.
+
+### Non-Goals
+
+Enable Cloudwatch metric collection.
+
+## Proposal
+
+### Cloudwatch streams and groups
+
+[Cloudwatch][concepts] defines *log groups* and *log streams*. To paraphrase the documentation:
+
+>  A log stream is a sequence of log events that share the same source ... For example, an Apache access log on a specific host.
+
+>  Log groups define groups of log streams that share the same retention, monitoring, and access control settings ... For example, if you have a separate log stream for the Apache access logs from each host, you could group those log streams into a single log group called MyWebsite.com/Apache/access_log.
+
+In other words a *log stream* corresponds to the smallest log-producing units.
+*Log groups* are used to group larger sets of related logs together.
+
+We must consider *container logs*, and *node logs*.
+- "application" logs are always container logs
+- "infrastructure" logs are a mix of container and node logs
+- "audit" logs are all node logs.
+
+For *container-scoped* logs we auto-create a log stream for *each container*, using the container UUID as the log stream name.
+- The log stream name is just a unique identifier for the source container.
+- Log *entries* include meta-data (container name, namespace etc.) for the user to search/index.
+  The log stream name is *not* needed for indexing or searching logs.
+- Log streams are for a *single source*, we cannot send logs from multiple nodes on the same log stream.
+
+For *node-scoped* logs we will create "audit"" and "infrastructure" streams for each node: e.g. `node-id/audit`.
+
+By default the forwarder creates 3 *log groups* for each of the log categories:  "application", "infrastructure" and "audit".
+Application logs can also be grouped by namespace, k8s labels, or forwarder configuration, see below.
+
+### API fields
+
+New API fields in the `output.cloudwatch` section:
+
+- `region`: (string) AWS region name, required to connect.
+- `retentionDays`: (number) Number of days to keep logs. Defaults to cloudwatch default retention.
+- `groupByNamespace`: (bool) If true, set the Cloudwatch group name to the source namespace.
+  If false, group name is set to the log category: "application", "infrastructure" or "audit".
+
+*Nice-to-have*: the following fields may be omitted from the first release and implemented later.
+
+- `group`: (string, *nice-to-have*) Fixed Cloudwatch log group name for all logs leaving this this output.
```

Comment:
`s/this this/this/`.

Can users query Cloudwatch logs across groups (looks like it is [available](https://aws.amazon.com/about-aws/whats-new/2019/07/cloudwatch-logs-insights-adds-cross-log-group-querying/))?  If so, then you might need to consider ways for the output stream to add metadata instead of changing the groups so that logs can be combined across groups.

Reply:
The forwarder does add all the metadata so everything is avaliable for query once cloudwatch has collected & stored the logs. This configuration is only for controlling how logs are *delivered* in terms of log streams and log groups.

Reply:
Metadata is available on all logs regardless of grouping. Whether/how the user choooses to use cloudwatch groups is a separate issue. I'm guessing that namespaces, labels or static groups will cover most use cases but I don't know cloudwatch enough to say what's the best way to group logs.